tmk_core/protocol/chibios/main.c

```diff
@@ -90,6 +90,10 @@ void console_task(void);
 void midi_ep_task(void);
 #endif
 
+#ifdef WAIT_US_TIMER
```

Comment:
wait_init would be implemented in chibios dir (and stubs in the other files), called in somewhere like <https://github.com/qmk/qmk_firmware/blob/master/tmk_core/common/keyboard.c#L278> so its init path is common

Reply:
I think it's safer to leave this in `main()` because the first thread is also the idle thread, which can't be stopped. If that changes then `main_thread` will need to be set differently in `main.c` instead of hiding in a `wait_init`.

lib/chibios/os/nil/src/ch.c: "The idle thread must not invoke any kernel primitive able to change state to not runnable"

I could remove the `#ifdef` but nothing else is using it?

Reply:
That gives a poor abstraction on the problem, you can want to leave it how it is, but in the end its just going to be PRed to update to a consistent interface anyway.

Reply:
I've updated it